app/src/main/java/com/treasurehunt/ui/BindingAdapters.kt

```diff
@@ -46,13 +49,15 @@ fun bindImageStorageUrl(view: ImageView, url: String?) {
                 .sizeMultiplier(0.25f)
             Glide.with(view.context)
                 .load(storageRef)
+                .error(R.drawable.ic_no_image)
                 .thumbnail(thumbnailRequestBuilder)
+                .diskCacheStrategy(DiskCacheStrategy.DATA)
```

Comment:
glide Cache 부분에 
AUTOMATIC으로 설정할때 remote data와 local data를 자동으로 적합하게 처리해준다는 내용이 있습니다.
(AUTOMATIC will store only the unmodified data backing your load when you’re loading remote data (like from URLs) because downloading remote data is expensive compared to resizing data already on disk. For local data AUTOMATIC will store the transformed thumbnail only because retrieving the original data is cheap if you need to generate a second thumbnail size or type.)
이걸 기반으로 로컬 이미지와 원격 이미지의 캐시 저장소를 확인했을때 로컬은 RESOURCE, 원격은 DATA 캐시에 저장하는걸 확인해서 DATA로 선택했습니다. 그리고 AUTOMATIC과 DATA를 를 비교했을때는 AUTOMATIC은 한번 저장한 캐시 위치를 변경해주지 않는데 DATA는 DATA 캐시로 위치를 이동시켜주기 때문에 더 나은 방법이라고 생각해서 선택했습니다.

Reply:
음 제가 잘 몰라서 그런 걸 수도 있는데 설명이 이해가 안 돼요
아래 링크 내용 보면,
DiskCacheStrategy.DATA -> 데이터를 받아와서 디코딩하기 전에 디스크 캐시에 저장
DiskCacheStrategy.RESOURCE -> 데이터를 디코딩해서 리소스 형태로 디스크 캐시에 저장
이렇게 나와 있어서 클래스명에도 나와 있듯이 "디스크 캐시"에 저장을 하는 걸로 보여지는데, 리소스 캐시랑 데이터 캐시라고 하신 부분이 일단 이해가 안 돼요
위에 인용하신 영문에는 그 내용 없는 거 같은데, 어디서 참고하셨는지 알 수 있을까요?
그리고 혹시 로그 통해 확인했다는 내용 스크린샷 같은 거 있으면 참고하고 싶습니다
https://bumptech.github.io/glide/javadocs/420/com/bumptech/glide/load/engine/DiskCacheStrategy.html

Reply:
<img width="189" alt="스크린샷 2024-02-14 오후 9 40 33" src="https://github.com/Sesac-team-03/TreasureHunt/assets/82762769/7c865087-2497-453e-89d2-e2290cab66ff">
<img width="206" alt="스크린샷 2024-02-14 오후 9 43 17" src="https://github.com/Sesac-team-03/TreasureHunt/assets/82762769/b5b775ff-7e71-4d17-969f-e49515be5baa">
정근님 말씀이랑 링크를 보니 로그를 찍어본 내용이 저장 위치가 아니라 저장 방식을 보여주는건가 싶기도한데요
저도 이 부분은 정확하지 않아서 공부를 좀 더 해봐야 할 것 같습니다!

Reply:
스크린샷을 저렇게 보면 어떤 건지 전혀 알 수 없는 거 같습니다
로그에서 참조한 값이 뭐였을까요?

Reply:
지금 저희 코드에서 사용하는 코드입니다.
Glide.with(view.context)
                .load(storageRef)
                .error(R.drawable.ic_no_image)
                .thumbnail(thumbnailRequestBuilder)
                .diskCacheStrategy(DiskCacheStrategy.DATA)
                .listener(object : RequestListener<Drawable> {
                    override fun onLoadFailed(
                        e: GlideException?,
                        model: Any?,
                        target: Target<Drawable>,
                        isFirstResource: Boolean
                    ): Boolean {
                        return false
                    }

                    override fun onResourceReady(
                        resource: Drawable,
                        model: Any,
                        target: Target<Drawable>?,
                        dataSource: DataSource,
                        isFirstResource: Boolean
                    ): Boolean {
                        // 이 부분 데이터 dataSource
                        Log.d( "디스크 캐시","$dataSource")
                        return false
                    }
                })
                .into(this)
이 부분에 dataSource가 데이터의 출처를 나타내게 되고 위 사진이 로그를 통해 출력한 내용입니다
이때 AUTOMATIC을 사용하면 REMOTE에서 불러오는 코드는 DATA_DISK_CACHE 출처를 가지게 되는걸 볼 수 있고 해당 이유로
리뷰에서 설명드린 부분을 다시보면 이해가 될 것 같습니다.
아니면 해당 코드를 통해 직접 찍어보시는게 이해가 더 잘 될수도 있을꺼 같아요.

Reply:
음 저희 코드에서 사용하는 코드라고 하셨는데 저건 브랜치에서 아무리 찾아봐도 없는데 listener를 추가해주신 거 맞을까요?
동현님 일단 캐시에 대해 잘못 설명하신 부분 제가 말씀드렸고, 잘못 이해하신 내용을 바탕으로 판단을 하셨기 때문에 그 판단이 유효하다고 설득하려면 다른 제대로 된 근거와 설명을 가져오셔야 합니다
리뷰에서 설명해주신 부분 뭘 다시 보라는 건지 전혀 모르겠어요
"AUTOMATIC을 사용하면 REMOTE에서 불러오는 코드는 DATA_DISK_CACHE 출처를 가지게 되는걸" -> 그리고 지금 이 내용도 틀렸습니다
AUTOMATIC을 사용하면 최초로 로딩할 때 REMOTE 데이터소스에서 불러오고, 그 다음 앱이 실행되는 동안 피드화면에 재진입하게 되면 MEMORY_CACHE에서 불러오고, 만약 앱이 종료된 후 재실행되어 피드화면에 진입하면 그때 DISK_CACHE에서 불러오는 것입니다
그리고 DataSource 내부코드 혹시 보셨나요?
https://bumptech.github.io/glide/javadocs/420/com/bumptech/glide/load/DataSource.html
위 링크의 설명만 봐도 DATA_DISK_CACHE는 캐시 이름이 아니라고 나와 있습니다
DISK_CACHE라는 것밖에 없구요
제가 이전 댓글에서도 이미 설명했지만, 원격에서 불러온 데이터를 그대로 저장하는 경우 Data, 디코딩해서 저장하는 경우 Resource로 구분해서 Datasource 이넘 클래스의 상수명이 각각을 반영하도록 지어진 것입니다

종합하자면 동현님이 아래와 같이 DATA를 선택한 이유라고 처음 얘기하신 부분은 내용 자체가 틀렸고 타당하다고 설득이 안 됩니다
"이걸 기반으로 로컬 이미지와 원격 이미지의 캐시 저장소를 확인했을때 로컬은 RESOURCE, 원격은 DATA 캐시에 저장하는걸
로그를 통해 확인해서 DATA로 선택했습니다. 그리고 AUTOMATIC과 DATA를 비교했을때는 AUTOMATIC은 한번 저장한 캐시 위치를 변경해주지 않는데 DATA는 DATA 캐시로 위치를 이동시켜주기 때문에 더 나은 방법이라고 생각해서 선택했습니다."

캐시 관련해서 다시 한번 공부하신 다음에 제가 말씀드린 부분 이해 안 가거나 틀린 내용 있으면 알려주세요
타당한 이유가 없다면 기본값인 Automatic이나 아니면 글라이드에서 권장한 Resource가 적합할 거 같습니다
Data를 사용하게 되면 로딩 속도와 디스크 공간을 낭비하는 단점이 있지만, 같은 리소스를 여러 사이즈로 여러 번 사용하는 경우 이점이 있다고 했는데, 피드화면에 해당되는 이점이라고 생각되지 않습니다

마지막으로 이 캐시 전략은 최초 로딩 시간 단축에 얼마나 유의미할진 모르겠으니 제가 앞서 디스코드에 말씀드린 preload 방법 찾아서 적용해주시면 감사하겠습니다